src/services/ui/src/components/Form/fields/AttachmentsSection.tsx

```diff
@@ -9,13 +9,20 @@ import {
   RequiredIndicator,
   Upload,
 } from "@/components";
+import { attachmentTitleMap } from "shared-types";
+import type { AttachmentKey } from "shared-types";
+
+type AttachmentType = {
+  name: AttachmentKey;
+  required: boolean;
+};
```

Comment:
this is a very similar shape to `AttachmentRecipe` which uses the generic to type the `name` attribute. I think we can skip the middleman and achieve typing in `AttachmentSection.name` and remove the generic entirely. let's try it today!